### DISCUSSION TITLE
[Feature] User permission hierarchical management

### DISCUSSION THREAD

**Discussion by bigsinger** [immich-app/immich#2871] [Feature Request]
### The feature

User permission hierarchical management. Is there a hierarchical management of user permissions, such as having one administrator and others as regular users. Ordinary users can only manage their own account's photo albums. Administrators can see their own albums, including those of other ordinary users, and can support switching perspectives at any time.

This application scenario is suitable for: company album management (supervisors need administrators), class album management (teachers need administrators), and home album management (parents need administrators).

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by echondropoulos** [comment]
Wanted to add onto this(since its similar and didn't want to make another Feature request thread), but I'm hoping for more granular control of users and what roles/permissions they'd have access to. And potentially more permissions too.

Example:

| Roles         | Permissions                             |
|---------------|-----------------------------------------|
| Administrator | All |
| Supervisor    | Create and Edit albums                  |
| Contributor   | Upload to albums                        |
| Guest        | View albums                             |

Or having the ability to create roles and permissions for each. For me, as an administrator, I should be able to see ALL photos and ALL albums and be able to edit and share ALL albums as well.

**Comment by MattKayDev** [comment]
Yes custom roles would work well for this, as administrator you want to see all pictures but when adding pictures to album you might want to select which role can view the photo
